napari/layers/labels/labels.py

```diff
@@ -1122,9 +1129,11 @@ def _update_thumbnail(self):
         )
         zoom_factor = tuple(new_shape / imshape)
 
-        downsampled = ndi.zoom(image, zoom_factor, prefilter=False, order=0)
+        downsampled = _convert_small_ints_to_unsigned(
+            ndi.zoom(image, zoom_factor, prefilter=False, order=0)
+        )
         if self.color_mode == LabelColorMode.AUTO:
-            color_array = self.colormap._map_precast(downsampled.ravel())
+            color_array = self.colormap.map(downsampled.ravel())
```

Comment:
I think `.ravel()` is actually no longer needed since #5805; same for below, and the subsequent `reshape`.